chroma-agent/chroma_agent/action_plugins/manage_node.py

```diff
@@ -75,4 +76,18 @@ def initialise_block_device_drivers():
 
     return agent_result_ok
 
+
+# When the agent is stopped we want to allow block devices to do any termination that they might need, this function
+# may also be called by the manager.
+@agent_daemon_teardown_function()
+def terminate_block_device_drivers():
+    console_log.info("Terminating drivers for block device types")
+    for cls in util.all_subclasses(BlockDevice):
+        error = cls.terminate_driver(config.profile_managed)
```

Comment:
on looking at this, we should probably revisit whether we use that flag in initialise_driver. at the moment . if monitored mode we don't stop the services (zfs.target/import-scan/import-cache/mount), that was a patch you implemented from the conversations we had with Don Brady. we should probably be doing that in monitored mode as we still do the import-export in scans. that change would obviously be for another patch: https://github.com/intel-hpdd/intel-manager-for-lustre/issues/106